crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -197,42 +196,40 @@ impl<'a, Provider: ResolverProvider> Resolver<'a, Provider> {
         let requests_fut = self.fetch(request_stream).fuse();
 
         // Run the solver.
-        let resolve_fut = self.solve(&request_sink).fuse();
+        let resolve_fut = self.solve(request_sink).fuse();
```

Comment:
Taking ownership means that `request_sink` is dropped as soon as the future completes.